float32[120] raycast
